cdap-docs/tools/doc-cdap-default.py

```diff
@@ -0,0 +1,434 @@
+    else:
+        exclusions = load_exclusions()
+    
+    items = []
+    parser = PIParser()
+    tree = ET.parse(xml_path, parser=parser)
+    root = tree.getroot()
+    for outer_element in root:
+        for inner_element in outer_element:
+            if inner_element.tag == 'property':
+                item = Item()
+                for attribute in inner_element:
+                    item.set_attribute(attribute.tag, attribute.text)                
+                if item.name not in exclusions:
+                    items.append(item)                
+            else:
+                items.append(Section(inner_element.text))
+
+    return items
+
+def create_rst(items):
+    """Create rst from items"""
+    table = ''
+    final_items =[]
+    
+    for item in items:
+        if not item.final:
+            table += item.rst() + '\n'
+        else:
+            final_items.append(item)
+            
+    if final_items:
+        table += RST_TABLE_HEADER_FINAL
+        for item in final_items:
+            table += item.rst() + '\n'
+
+    return table
+
+def rebuild():
+    """Loads the cdap-default.xml, and rebuilds it according to these rules:
+    - First two sections: 
+        General Configuration
+        Global Configuration
+    - Remaining sections alphabetical
+    - Properties within a section alphabetical
+    - Descriptions wrapped to a 70 character line-length"""
+    
+    print 'Rebuilding cdap-default.xml'
+    items = load_defaults('', include_exclusions=True)
+    exclusions = load_exclusions()
+    in_section = True
+    defaults = {}
+    section = []
+    section_counter = 0
+    properties_counter = 0
+    for item in items:
+        if str(item.__class__) == '__main__.Section':
+            if section: # End old section
+                defaults[section_name] = section
+                section = []
+            # Start new section
+            section_counter += 1
+            section_name = item.name
+        else:
+            properties_counter += 1
+            section.append(item)
+    if section: # End old section
+        defaults[section_name] = section
+        section = []  
+    
+    print "Sections: %d Keys: %d Properties: %d" % (section_counter, len(defaults.keys()), properties_counter)
+
+    # Build XML file
+    XML_CONFIG_OPEN   = '<configuration>\n'
+    XML_SECTION_SUB   = "\n    <!-- %s -->\n\n"
+    XML_PROP_OPEN     = '    <property>\n'
+    XML_NAME_SUB      = "        <name>%s</name>\n"
+    XML_VALUE_SUB     = "        <value>%s</value>\n"
+    XML_DESCRIP_OPEN  = '        <description>\n'
+    XML_DESCRIP_SUB   = "            %s\n"
+    XML_DESCRIP_CLOSE = '        </description>\n'
+    XML_FINAL         = '        <final>true</final>\n'
+    XML_PROP_CLOSE    = '    </property>\n\n'
+    XML_CONFIG_CLOSE  = '</configuration>\n'
+
+    xml = XML_CONFIG_OPEN
+    keys = defaults.keys()
+    for key in FIRST_TWO_SECTIONS:
+        keys.remove(key)
+    keys.sort()        
```

Comment:
nit: this (or line 330) now makes hdfs.lib.dir (which references hdfs.namespace) appear before hdfs.namespace :-)